pymc3/distributions/multivariate.py

```diff
@@ -522,7 +522,6 @@ def logp(self, value):
             tt.sum(logpow(value, a - 1) - gammaln(a), axis=-1) + gammaln(tt.sum(a, axis=-1)),
             tt.all(value >= 0),
             tt.all(value <= 1),
-            np.logical_not(a.broadcastable),
```

Comment:
I just removed the entire broadcasting logic to handle any shape for `a` because now I think, there are no restrictions on shape parameter while calculating logp.

Reply:
Yeah, as long as the log-likelihood calculations can handle the broadcasting themselves, there's no need for that condition.

I still think that a `k > 1` input validation step in `Dirichlet.__init__` wouldn't hurt, but it doesn't really matter when our broadcasting logic is sound.